### PR TITLE
Return valid json in Server fixture

### PR DESCRIPTION
**Pull request by heeckhau**
* return json instead of string
* removed trailing whitespace
* use a constant for the default port
* give binary a better name

### CODE REVIEW

File: crates/server-fixture/server/Cargo.toml [hunk 1]
```diff
@@ -14,9 +14,14 @@ hyper-util = { workspace = true, features = ["full"] }
...
+serde_json = { workspace = true }
 tlsn-server-fixture-certs = { workspace = true }
 
 [[bin]]
-name = "main"
+name = "tlsn-server-fixture"
 path = "src/main.rs"
+
+[dev-dependencies]
+axum-test = "16.2.0"
```
Comment (yuroitaki) [inline crates/server-fixture/server/Cargo.toml:25]: nit: i think we're adopting the format of always using `{ version = "16.2.0" }` in the project?

File: crates/server-fixture/server/src/lib.rs [hunk 2]
```diff
@@ -94,10 +97,18 @@ async fn bytes(
...
+        eprintln!("Failed to parse JSON data: {}", e);
+        StatusCode::INTERNAL_SERVER_ERROR
+    })?))
+}
+
 async fn json(
     State(state): State<Arc<Mutex<AppState>>>,
     Query(params): Query<HashMap<String, String>>,
-) -> Result<Json<&'static str>, StatusCode> {
+) -> Result<Json<Value>, StatusCode> {
```
Comment (yuroitaki) [inline crates/server-fixture/server/src/lib.rs:111]: Does any downstream client using this server fixture need to change their handling of the response?

File: crates/server-fixture/server/src/lib.rs [hunk 3]
```diff
@@ -125,3 +136,58 @@ async fn html(
...
+        let state = AppState {
+            shutdown: Some(sender),
+        };
+        app(state)
+    }
+
+    #[tokio::test]
+    async fn hello_world() {
+        let response = get_app()
+            .oneshot(Request::builder().uri("/").body(Body::empty()).unwrap())
```
Comment (yuroitaki) [inline crates/server-fixture/server/src/lib.rs:161]: this definitely worked for testing purely the request and response; tho if we want more of an integration test of the server, then the test should spin up the server like its `main.rs` (using `tokio::spawn` & `TcpListener`)

**Review by yuroitaki — COMMENTED** [review]
nice stuff!

File: crates/server-fixture/server/Cargo.toml (hunk 1 above)
Comment (heeckhau) [reply crates/server-fixture/server/Cargo.toml]: fixed

File: crates/server-fixture/server/src/lib.rs (hunk 2 above)
Comment (heeckhau) [reply crates/server-fixture/server/src/lib.rs:111]: All tests are green. So should be fine

**Pull request merged**